### ISSUE TITLE
Spike: OLRC classification tables as ground truth for amendment-to-provision matching

### ISSUE THREAD

**Issue by Scronkfinkle** [Scronkfinkle/words-to-data#126]
> *This was generated by AI during triage.*

## Why

`match-amendments` asks a language model which provision an amendment changed. The Office of Law Revision Counsel **publishes the answer**, for the same Code it publishes the USLM files this project already ingests.

<https://usc-cdn.house.gov/classification/tables.shtml>

The eventual goal is a model that can predict changes forward and reconstruct them backward. That goal is not served by having a model guess at something an authority has already written down. A published table is a heuristic, a validation set, and eventually training data. It should be exhausted before a model is asked.

This is a spike. We keep the knowledge and throw away most of the code.

## What the tables actually contain

Verified during triage. Real rows from the 119th Congress, 1st session table:

```
U. S. Code Title  Section   Description   Pub. L.   Sec.              139 Stat.
8                 1101      nt new        119-1     1                 [3]
26                36B                     119-21    71301(a), (b)     [321]
42                1396a                   119-21    71103(a)(1)       [291]
5                 8905      nt new        119-21    90101(a)-(f)      [362]
38                101       nt new        119-31    1                 [475]
```

So each row gives the Code title and section affected, **the kind of change** (`new`, `nt new`, amended where blank, repealed), the public law, **the section within that law**, and the Statutes page.

The section within the law is the part that matters: it identifies the *amending provision*, which is exactly what this project calls an amendment.

- **Two sort orders** are published per session: public law order (`tbl119pl_1st.htm`) and Code order (`tbl119cd_1st.htm`). Presumably the same facts, indexed both ways — confirm that.
- **HTML only.** No CSV, no XML, no JSON. Parsing is the engineering cost.
- The page covers the 119th Congress. Prior tables are said to be available elsewhere; **finding out where, and how far back, is part of this spike.**

## The validation set is perfect, and it is already in the dataset

Every one of the **893** links in the real dataset comes from one bill, `119-hr-1`. That bill is **Public Law 119-21**, which appears throughout the 1st-session table.

So the spike can measure a published authority against 893 model-suggested links, on real data, with no new ingest.

**The join key already exists and is being discarded.** The cached bill metadata holds it:

```json
"laws": [ { "number": "119-21", "type": "Public Law" } ]
```

`download_bill` fetches that metadata and `load_bill_download` reads only the sponsors out of it. Nothing extracts the public law number, so nothing can currently join a bill to a classification table. Pulling it out is small.

Worth noting while you are there: `Bill.bill_id`'s doc comment says it holds `"119-21" for the 119th Congress, 21st law`, but the value actually passed in is `119-hr-1`. The comment describes an intent the code does not implement.

## The honest limit, which must be in the write-up

**The tables resolve to a Code section, not below it.** A row says `26 36B`; it does not say which subsection, paragraph or clause moved.

#106 measured where this project's links actually sit:

| | count | share |
| --- | --- | --- |
| on the section itself | 44 | 4.9% |
| below the section | 849 | **95.1%** |

So a classification table **cannot** replace the model for 95% of the corpus. What it can do is narrow the candidate set from a whole title to one section before any model call, and corroborate or contradict a match after one. Both are large wins. Neither is resolution.

A second limit, in the OLRC's own words: *"The tables only include those provisions of law that have been classified to the Code."* **Absence from a table is not absence of a change.** In this project's vocabulary that is a Gap, and it must be reported as one rather than read as "nothing happened".

## What to investigate

1. **Format and stability.** Parse one session's table. Are the two sort orders the same facts? Is the HTML stable enough to parse, or does it vary by session? Record the shape you found, not the shape you hoped for.
2. **Historical reach.** Find the prior-Congress tables and say how far back they go. The value of this work scales with coverage.
3. **The vocabulary of the Description column.** Enumerate every value that appears — `new`, `nt new`, blank, and whatever else — and say what each means. A mapping onto this project's `AmendingAction` is the useful output.
4. **Measure it against PL 119-21.** The pass condition below.
5. **Decide what a table match is worth.** `VerificationState::Asserted` means "a source says so", and the OLRC is a source. So a corroborated link could stand at `Asserted` rather than `MachineSuggested`. Recommend a mapping; the decision is the maintainer's. Note that `is_human_touched` must keep protecting human review either way (ADR 0004).

## Pass condition

For Public Law 119-21, report all four numbers:

- how many of the 893 stored links the table **corroborates**,
- how many it **contradicts**,
- how many it **cannot speak to**, separated into "the table is silent" and "the link is below section granularity",
- and what fraction of the model's candidate set the table **could have pruned before any model call**.

The last number is the one that decides whether this becomes production work. It also bears directly on #123: `match-amendments` has no cache, so every regenerate re-buys around 632 model replies. Pruning candidates cuts that bill at the source.

## Notes

- Do not build the production ingest. Do not add a mirror. Do not change `match-amendments`.
- Do not scrape more than the tables you need. This is a `.gov` CDN serving static HTML; be unhurried and cache what you fetch.
- Write the findings to `docs/research/olrc-classification-tables.md`, in the manner of `docs/research/courtlistener-formats.md`.
- If the spike makes you want to build the ingest, that is a finding to record and a follow-up issue to file, not work to do here.
- Related: #123 (no cache, so matching is re-bought), #106 (where links actually sit), #52 (U.S.C. citation extraction, which a Code-order table would feed).

## Done when

The four numbers above are reported for PL 119-21, `docs/research/olrc-classification-tables.md` records the format, the historical reach and the Description vocabulary, and we can say whether a classification table belongs in the pipeline as a pruner, as a corroborator, as both, or not at all.
